src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8333,6 +8376,280 @@ private StatementSyntax ParseLocalDeclarationStatement()
+
+            if (this.CurrentToken.IsVar && this.PeekToken(1).Kind == SyntaxKind.OpenParenToken)
+            {
+                // parses `var (...) = expression` form
+                return ParseDeconstructionVarForm(withEquals);
+            }
+            else if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken)
+            {
+                // parses `(...) = expression` form
+                return ParseDeconstructionList(withEquals, justIdentifiers: false);
+            }
+            else if (!topLevel)
+            {
+                // parses `type id` part
+                return ParseDeconstructionTypeIdPart();
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Parses `(..., ..., ...)` where each part is deconstruction variables (such as `type id` or `var (..., ...)` or another list `(..., ..., ...)`).
+        /// </summary>
+        private VariableDeclarationSyntax ParseDeconstructionList(bool withEquals, bool justIdentifiers)
+        {
+            if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken)
+            {
+                var list = _pool.AllocateSeparated<VariableDeclarationSyntax>();
+
+                try
+                {
+                    var openParen = this.EatToken(SyntaxKind.OpenParenToken);
+
+                    if (this.CurrentToken.Kind != SyntaxKind.CloseParenToken)
+                    {
+                        VariableDeclarationSyntax variable;
+                        while (true)
+                        {
+                            if (justIdentifiers)
+                            {
+                                variable = ParseDeconstructionIdentifierOrIdentifiersParts(withEquals: false);
+                            }
+                            else
+                            {
+                                variable = ParseDeconstructionDeclaration(withEquals: false, topLevel: false);
+                            }
+
+                            if (variable != null)
+                            {
+                                list.Add(variable);
+                            }
+
+                            if (this.CurrentToken.Kind != SyntaxKind.CommaToken)
+                            {
+                                break;
+                            }
+
+                            var comma = this.EatToken(SyntaxKind.CommaToken);
+                            list.AddSeparator(comma);
+                        }
+                    }
+
+                    var closeParen = this.EatToken(SyntaxKind.CloseParenToken);
+
+                    SyntaxToken equals = null;
+                    ExpressionSyntax expression = null;
+                    if (withEquals)
+                    {
+                        equals = this.EatToken(SyntaxKind.EqualsToken);
+                        if (equals.IsMissing)
+                        {
+                            return null;
```

Comment:
I am not sure we always want to abandon what we have parsed already. I think usually we add missing expression in situations like this.